apps/api/src/domain/models/song/commands/publish-song.command.integration.spec.ts

```diff
@@ -10,22 +10,37 @@ import { IIdManager } from '../../../interfaces/id-manager.interface';
 import { ResourceType } from '../../../types/ResourceType';
 import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
 import InvalidCommandPayloadTypeError from '../../shared/common-command-errors/InvalidCommandPayloadTypeError';
+import { assertCommandError } from '../../__tests__/command-helpers/assert-command-error';
+import { assertCommandSuccess } from '../../__tests__/command-helpers/assert-command-success';
+import { CommandAssertionDependencies } from '../../__tests__/command-helpers/types/CommandAssertionDependencies';
 import buildDummyUuid from '../../__tests__/utilities/buildDummyUuid';
+import { Song } from '../song.entity';
 import { PublishSong } from './publish-song.command';
 import { PublishSongCommandHandler } from './publish-song.command-handler';
 
 const publishSongCommandType = 'PUBLISH_SONG';
 
 const dummyUuid = buildDummyUuid();
 
-const validCommandFSA: FluxStandardAction<DTO<PublishSong>> = {
+const unpublishedSong = getValidResourceInstanceForTest(ResourceType.song).clone({
+    id: dummyUuid,
+    published: false,
+});
+
+const buildCommandFSA = (_): FluxStandardAction<DTO<PublishSong>> => ({
     type: publishSongCommandType,
     payload: {
-        id: dummyUuid,
+        id: unpublishedSong.id,
     },
-};
+});
+
+const buildFullState = (song: Song) =>
```

Comment:
It might be nice to have a helper that curries building this function once you give it the `ResourceType`.